utils/usage/usage_test.go

```diff
@@ -129,6 +131,64 @@ func createArtifactoryUsageHandler(t *testing.T, productName, commandName string
 	}
 }
 
+func TestReportXrayUsage(t *testing.T) {
+	const productName = "test-product"
+	const commandName = "test-command"
+	const clientName = "test-client"
+
+	server := httptest.NewServer(createXrayUsageHandler(t, productName, commandName, clientName))
+	defer server.Close()
+	serverDetails := &config.ServerDetails{XrayUrl: server.URL + "/"}
+
+	reporter := NewUsageReporter(productName, serverDetails).SetSendToEcosystem(false).SetSendToArtifactory(false)
+
+	reporter.Report(ReportFeature{
+		FeatureId: commandName,
+		ClientId:  clientName,
+	})
+	assert.NoError(t, reporter.WaitForResponses())
+}
+
+func TestReportXrayError(t *testing.T) {
+	reporter := NewUsageReporter("", &config.ServerDetails{}).SetSendToEcosystem(false).SetSendToArtifactory(false)
+	reporter.Report(ReportFeature{
+		FeatureId: "",
+	})
+	assert.Error(t, reporter.WaitForResponses())
+
+	server := httptest.NewServer(create404UsageHandler(t))
+	defer server.Close()
+	reporter = NewUsageReporter("", &config.ServerDetails{ArtifactoryUrl: server.URL + "/"}).SetSendToEcosystem(false).SetSendToArtifactory(false)
+	reporter.Report(ReportFeature{
+		FeatureId: "",
+	})
+	assert.Error(t, reporter.WaitForResponses())
+}
+
+func createXrayUsageHandler(t *testing.T, productId, commandName, clientId string) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.RequestURI == "/api/v1/system/version" {
+			w.WriteHeader(http.StatusOK)
+			_, err := w.Write([]byte(`{"xray_version":"6.9.0"}`))
+			assert.NoError(t, err)
+			return
+		}
+		if r.RequestURI == "/api/v1/usage/events/send" {
+			// Check request
+			buf := new(bytes.Buffer)
+			_, err := buf.ReadFrom(r.Body)
+			assert.NoError(t, err)
+			featureId := xrayusage.GetExpectedXrayEventName(productId, commandName)
+			assert.Equal(t, fmt.Sprintf(`[{"data":{"clientId":"%s"},"product_name":"%s","event_name":"%s","origin":"API_CLI"}]`, clientId, productId, featureId), buf.String())
+
+			// Send response OK
+			w.WriteHeader(http.StatusOK)
+			_, err = w.Write([]byte("{}"))
+			assert.NoError(t, err)
+		}
```

Comment:
Let's check that there are no unexpected requests
```suggestion
			return
		}
		assert.Fail(t, "Unexpected request URI", r.RequestURI)
```